activerecord/lib/active_record/model_schema.rb

```diff
@@ -435,15 +439,27 @@ def reset_column_information
         initialize_find_by_cache
       end
 
+      protected
+
+        def initialize_load_schema_monitor
+          @load_schema_monitor = Monitor.new
+        end
+
       private
 
+        def inherited(child_class)
+          super
+          child_class.initialize_load_schema_monitor
```

Comment:
Ok, now I see what you meant when you said "class-local" in https://github.com/rails/rails/pull/28756#issuecomment-294162102 - each model getting its own Monitor (vs a single Monitor shared by all subclasses of `ActiveRecord::Base` in https://github.com/rails/rails/pull/28756).